libdnf/goal/Goal.cpp

```diff
@@ -737,37 +695,33 @@ Goal::countProblems()
 DnfPackageSet *
 Goal::listConflictPkgs(DnfPackageState pkg_type)
 {
-    DnfPackageSet *pset = dnf_packageset_new(pImpl->sack);
-    DnfPackageSet *temporary_pset = dnf_packageset_new(pImpl->sack);
+    DnfSack * sack = pImpl->sack;
+    Pool * pool = dnf_sack_get_pool(sack);
+    DnfPackageSet *pset = dnf_packageset_new(sack);
+    PackageSet temporary_pset(sack);
 
     int countProblemsValue = pImpl->countProblems();
     for (int i = 0; i < countProblemsValue; i++) {
-        Queue *conflict = pImpl->conflictPkgs(i);
-        for (int j = 0; j < conflict->count; j++) {
-            Id id = conflict->elements[j];
-            DnfPackage *pkg = dnf_package_new(pImpl->sack, id);
-            if (!pkg)
-                continue;
-            if (pkg_type ==  DNF_PACKAGE_STATE_AVAILABLE && dnf_package_installed(pkg)) {
-                dnf_packageset_add(temporary_pset, pkg);
+        auto conflict = pImpl->conflictPkgs(i);
+        for (int j = 0; j < conflict->size(); j++) {
+            Id id = conflict->get(j);
+            Solvable *s = pool_id2solvable(pool, id);
+            bool installed = pool->installed == s->repo;
+            if (pkg_type ==  DNF_PACKAGE_STATE_AVAILABLE && installed) {
+                temporary_pset.set(id);
                 continue;
             }
-            if (pkg_type ==  DNF_PACKAGE_STATE_INSTALLED && !dnf_package_installed(pkg))
+            if (pkg_type ==  DNF_PACKAGE_STATE_INSTALLED && !installed)
                 continue;
-            dnf_packageset_add(pset, pkg);
+            pset->set(id);
         }
-        queue_free(conflict);
     }
-    unsigned int count = dnf_packageset_count(temporary_pset);
-    if (!count) {
-        delete temporary_pset;
+    if (!temporary_pset.size()) {
         return pset;
     }
 
-    DnfPackageSet *final_pset = remove_pkgs_with_same_nevra_from_pset(pset, temporary_pset,
-                                                                      pImpl->sack);
+    DnfPackageSet *final_pset = remove_pkgs_with_same_nevra_from_pset(pset, &temporary_pset, sack);
```

Comment:
if unique_ptr is used for pset then 
return remove_pkgs_with_same_nevra_from_pset(pset, &temporary_pset, sack);